src/glomap/math/gravity.cc

```diff
@@ -24,66 +24,42 @@ Eigen::Matrix3d GetAlignRot(const Eigen::Vector3d& gravity) {
   return R;
 }
 
-double RotUpToAngle(const Eigen::Matrix3d& R_up) {
-  return colmap::RotationMatrixToAngleAxis(R_up)[1];
+double YawFromRotation(const Eigen::Matrix3d& rotation) {
+  return colmap::RotationMatrixToAngleAxis(rotation)[1];
 }
 
-Eigen::Matrix3d AngleToRotUp(double angle) {
-  return colmap::AngleAxisToRotationMatrix(Eigen::Vector3d(0, angle, 0));
+Eigen::Matrix3d RotationFromYaw(double yaw) {
+  return colmap::AngleAxisToRotationMatrix(Eigen::Vector3d(0, yaw, 0));
 }
 
-// Code adapted from
-// https://gist.github.com/PeteBlackerThe3rd/f73e9d569e29f23e8bd828d7886636a0
-Eigen::Vector3d AverageGravity(const std::vector<Eigen::Vector3d>& gravities) {
-  if (gravities.size() == 0) {
-    std::cerr
-        << "Error trying to calculate the average gravities of an empty set!\n";
+Eigen::Vector3d AverageGravityDirection(
```

Comment:
Maybe we generalize this function and call it ComputeAverageDirection or similar. We also already have ComputeAverageQuaternion on the colmap side, which does essentially the same thing but in 4d. Something for a follow-up PR.